source/common/network/tcp_listener_impl.cc

```diff
@@ -40,8 +40,9 @@ void TcpListenerImpl::onSocketEvent(short flags) {
   ASSERT(bind_to_port_);
   ASSERT(flags & (Event::FileReadyType::Read));
 
-  // TODO(fcoras): Add limit on number of accepted calls per wakeup
-  while (1) {
+  uint32_t connections_accepted_count = 0;
+  for (; connections_accepted_count < max_connections_to_accept_per_socket_event_;
+       ++connections_accepted_count) {
```

Comment:
When we reach the `max_connections_to_accept_per_socket_event_`, the next event loop only can be triggered by new socket event. If there is no new event, then the left connection won't be processed until a new event arrives(or never be processed if there is no new event) since we are using the edge trigger event.

To prevent this case, we should activate a read event manual after reaching `max_connections_to_accept_per_socket_event_`

Reply:
emm...thanks, I assume we are using edge-triggered in all places, I'm wrong here.